File: app/src/main/java/com/imangazalievm/bubbble/presentation/global/ui/views/dribbbletextview/TouchableUrlSpan.java
package com.imangazalievm.bubbble.presentation.global.ui.views.dribbbletextview;

import android.content.res.ColorStateList;
import android.graphics.Color;
import android.text.TextPaint;
import android.text.style.URLSpan;

public class TouchableUrlSpan extends URLSpan {

    private static int[] STATE_PRESSED = new int[]{android.R.attr.state_pressed};
    private boolean isPressed;
    private int normalTextColor;
    private int pressedTextColor;

    public TouchableUrlSpan(String url,
                            ColorStateList textColor) {
        super(url);
        this.normalTextColor = textColor.getDefaultColor();
        this.pressedTextColor = textColor.getColorForState(STATE_PRESSED, normalTextColor);
    }

    public void setPressed(boolean isPressed) {
        this.isPressed = isPressed;
    }

    @Override
    public void updateDrawState(TextPaint drawState) {
        drawState.setColor(isPressed ? pressedTextColor : normalTextColor);
        drawState.bgColor = Color.TRANSPARENT;
        drawState.setUnderlineText(!isPressed);
    }

}

File: app-mvp/core-ui/src/main/java/com/bubbble/coreui/ui/views/dribbbletextview/TouchableUrlSpan.kt
package com.bubbble.coreui.ui.views.dribbbletextview

import android.R
import android.content.res.ColorStateList
import android.graphics.Color
import android.text.style.URLSpan
import android.text.TextPaint

open class TouchableUrlSpan(
    url: String?,
    textColor: ColorStateList
) : URLSpan(url) {

    private var isPressed = false
    private val normalTextColor: Int = textColor.defaultColor
    private val pressedTextColor: Int = textColor.getColorForState(STATE_PRESSED, normalTextColor)

    fun setPressed(isPressed: Boolean) {
        this.isPressed = isPressed
    }

    override fun updateDrawState(drawState: TextPaint) {
        drawState.color = if (isPressed) pressedTextColor else normalTextColor
        drawState.bgColor = Color.TRANSPARENT
        drawState.isUnderlineText = !isPressed
    }

    companion object {
        private val STATE_PRESSED = intArrayOf(R.attr.state_pressed)
    }

}